File: Codeforces/702C - Cellular Network.cpp
// ==================================================
// Problem  :   702C - Cellular Network
// Run time :   0.062 sec.
// Language :   C++11
// ==================================================

#include <cstdio>
#include <algorithm>
#include <cmath>
using namespace std;

typedef     long long       LL;

int main()
{
    int n, m;

    while(~scanf("%d %d", &n, &m)) {
        LL cities[n];
        for(int i = 0; i < n; ++i) scanf("%I64d", cities+i);

        LL towers[m];
        for(int i = 0; i < m; ++i) scanf("%I64d", towers+i);

        int j = 0;
        LL ans = 0;

        for(int i = 0; i < n; ++i) {
            while(j+1 < m && abs(cities[i] - towers[j]) >= abs(cities[i] - towers[j+1])) ++j;
            ans = max(ans, abs(cities[i] - towers[j]));
        }

        printf("%I64d\n", ans);
    }

    return 0;
}

File: Codeforces/702C - Cellular Network.py
# ==================================================
# Problem  :   702C - Cellular Network
# Run time :   0.264 sec.
# Language :   Python 3
# ==================================================

def main():
	from sys import stdin, stdout
	
	inp = stdin.read().split()

	n = int(inp.pop(0))
	m = int(inp.pop(0))

	cities = [int(x) for x in inp[0:n]]	
	towers = [int(x) for x in inp[n:(n+m)]]

	j, ans = 0, 0

	for ct in cities:
		while j+1 < m and abs(ct - towers[j]) >= abs(ct - towers[j+1]):
			j += 1
		ans = max(ans, abs(ct - towers[j]))

	stdout.write(str(ans) + '\n')

if __name__ == '__main__':
	main()
